pkg/controller/job/job_controller.go

```diff
@@ -85,6 +79,10 @@ var (
 	MaxPodCreateDeletePerSync = 500
 )
 
+// MaxFailedIndexesExceeded indicates that an indexed of a job failed
```

Comment:
@alculquicondor I moved the MaxFailedIndexesExceeded to the controller and removed it from API.  

I think the other reasons should be fine to include in the API.

Reply:
updated.